annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -113,25 +116,73 @@ private boolean annotationNeeded(MutationRecord record) {
     public AnnotatedRecord annotateRecord(MutationRecord mRecord, boolean replace, String isoformOverridesSource, boolean reannotate)
             throws GenomeNexusAnnotationFailureException
     {
-
+        AnnotatedRecord annotatedRecord = new AnnotatedRecord(mRecord);
         //check if record already is annotated
         if(!reannotate && !annotationNeeded(mRecord)) {
-            return new AnnotatedRecord(mRecord);
+            return annotatedRecord;
         }
 
         String genomicLocation = this.extractGenomicLocationAsString(mRecord);
         VariantAnnotation gnResponse = null;
         try {
             gnResponse = this.apiClient.fetchVariantAnnotationByGenomicLocationGET(genomicLocation,
                     isoformOverridesSource,
+                    "",
                     Arrays.asList(this.enrichmentFields.split(",")));
         } catch (ApiException e) {
-            throw new GenomeNexusAnnotationFailureException("Empty Response From Genome Nexus: " + genomicLocation);
+            // catch case where Genome Nexus Server is down
+            // not logging here because if GN is down you could write out an arbitarily large logfile of "failures"
+            throw new GenomeNexusAnnotationFailureException("Server error from Genome Nexus: " + genomicLocation);
+        }
+        // catch case where annotation fails (server will return default "failed" variant) 
+        if (!gnResponse.isSuccessfullyAnnotated()) {
+            // only logs cases which can't be annotated due to a problem with input
+            LOG.warn("Annotation failed for variant " + gnResponse.getVariant());
+            throw new GenomeNexusAnnotationFailureException("Genome Nexus failed to annotate: " + gnResponse.getVariant());
         }
-
         return convertResponseToAnnotatedRecord(gnResponse, mRecord, replace);
     }
 
+    public List<AnnotatedRecord> annotateRecordsUsingGET(AnnotationSummaryStatistics summaryStatistics, List<MutationRecord> mutationRecords, String isoformOverridesSource, Boolean replace) {
```

Comment:
I appreciate that this code now lives in the annotator and not in the reader anymore - nice work.